hardhat.config.ts

```diff
@@ -73,7 +72,7 @@ const config: HardhatUserConfig = {
             accounts: {
                 mnemonic: process.env.MNEMONIC || HARDHAT_NETWORK_MNEMONIC,
             },
-            ...((isNativeChainType || isLocalFork) && bellecourNetworkConfig),
+            ...bellecourBaseConfig,
```

Comment:
Changed it be we have been testing on `Shagnhai` fork.

Reply:
I find it risky to have most of our tests running in Token mode while our main target is a Native mode so I wanted to switch it. I found it a good opportunity because I had a problem with the ZeroAddress not having funds to send transactions (with impersonation mechanism) but I understand that it may not be suitable for this PR so I reverted that change and used another fix. However, I still think we need to think about the subject.